source/fundamentals/bson.txt

```diff
@@ -0,0 +1,301 @@
+=================
+Working with BSON
+=================
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn about how the Go Driver handles conversions
+between BSON and Go objects. The process of converting a Go object to
+BSON is called **marshalling**, while the reverse process is called **unmarshalling**. 
+
+You should read this guide if you want to learn more about how the Go Driver
+represents BSON data or need to adjust default marshalling
+and unmarshalling behaviors.
+
+Data Types
+----------
+
+MongoDB stores documents in a binary representation called :manual:`BSON
+</reference/bson-types/>` that allows for easy and flexible data processing. The Go
+Driver represents BSON data using two families of types: ``D`` and ``Raw``.
```

Comment:
The description of "two families of types" doesn't really make sense in the context of how the `D`/`E`/`M`/`A` and `Raw` types are used.

This should be something like:
```
The Go Driver provides four main types for working with BSON data:

- ``D``: An ordered representation of a BSON document
- ``E``: Represents a single element in a BSON document
- ``M``: An unordered representation of a BSON document
- ``A``: An ordered representation of a BSON array
```

The `Raw` type is mostly relevant to unmarshaling and probably doesn't need to be described here.

Reply:
got it. omitted info about Raw types and changed the list intro to your suggestion!